fab/operations/supervisor.py

```diff
@@ -72,14 +73,41 @@ def set_celery_supervisorconf():
         'logistics_background_queue':   ['supervisor_celery_logistics_background_queue.conf'],
         'async_restore_queue':          ['supervisor_celery_async_restore_queue.conf'],
         'flower':                       ['supervisor_celery_flower.conf'],
-        }
+    }
 
     queues = _get_celery_queues()
+    if 'periodic' in queues and env.host != queues['periodic'].get('server_whitelist'):
+        show_periodic_server_whitelist_message_and_abort(env)
     for queue, params in queues.items():
         for config_file in conf_files[queue]:
             _rebuild_supervisor_conf_file('make_supervisor_conf', config_file, {'celery_params': params})
 
 
+def show_periodic_server_whitelist_message_and_abort(env):
+    utils.abort(
+        "\n\n"
+        "You're seeing this message as part of an effort to reduce the chance of\n"
+        "deploying celery beat, which sends SMSes, fires off reminders,\n"
+        "and triggers forwarders, among other things, unintentionally.\n\n"
+        "For example, during a migration, you initially want to deploy everything\n"
+        "except for celery beat, because you don't want both the old and new servers\n"
+        "running celery beat; that screws up queuing, etc.\n\n"
```

Comment:
now that i think about it, if we also implemented the same kind of thing for these services i think we could have also avoided the virginia issue:
https://github.com/dimagi/commcare-hq-deploy/blob/55e06de71100187b7c607c4ec90917496bb51edd/fab/operations/supervisor.py#L153-L171